### PR TITLE
Do not change the buffer if indentation is correct

### PR DESCRIPTION
**Pull request by Hi-Angel**
When indenting the buffer, the plugin currently always modifies it, even when there's nothing to modify. Fix that.

### CODE REVIEW

**Comment by Hi-Angel** [comment]
Actually, my description turns out a bit wrong, the `indent-to` does not modify buffer when it's not needed. It's just that before using it the `delete-region` is being called to remove the indentation.

Let me fix the description.

**Comment by Hi-Angel** [comment]
**UPD**: found a bug in the Makefile, added a commit fixing it.

On related note, is it perhaps better to use `ninja` instead of old Makefile? It is similarly a low-level build-system, but its syntax is much less involved, so it's just easier to maintain. For an example of a Emacs plugin where Ninja is used for testing see [here](https://github.com/ankurdave/color-identifiers-mode/blob/master/build.ninja). It's [faster too](https://neugierig.org/software/chromium/notes/2011/02/ninja.html), although here I presume it won't be noticeable.

**Comment by Hi-Angel** [comment]
ping

**Comment by Hi-Angel** [comment]
Ping

**Comment by Hi-Angel** [comment]
Ping

**Review by tgross — APPROVED** [review]
LGTM! Sorry about the delayed review

**Comment by Hi-Angel** [comment]
Np, and thank you for looking into it!

**Comment by tgross** [comment]
Fixes: https://github.com/hcl-emacs/hcl-mode/issues/7

It turns out I don't have merge-approvals on this repo though :blush: , so we'll need @Fuco1 or @purcell to click the merge button.

**Comment by purcell** [comment]
Merged. And I think I've bumped your privileges so that you have admin access to the repos here, @tgross.

**Comment by Hi-Angel** [comment]
@purcell hello, sorry to bother you, I just stumbled upon this old thread and figured you gave permissions to @tgross. I understand you wanted to boost up the maintenance, but I'm afraid you gave permissions to very wrong person. Not only he has never contributed to the mode, but I just looked at his contributions and [he has never had Emacs-related contributions](https://github.com/search?q=author%3Atgross+emacs&type=code).

image

For 9 months of having permissions his single contribution was [shutting down the contributor here](https://github.com/hcl-emacs/hcl-mode/pull/12). I think he's had enough time to show his nature and it's time to revoke the rights.

-------------

OTOH, I can help with co-maintaining the mode. Not promising to do much as I don't use the mode right now *(ATM I am on a different project at work)*, but at least I could review PRs. I'm a co-maintainer to [color-indentifiers-mode](https://github.com/ankurdave/color-identifiers-mode) and I've been [occasionally contributing to Emacs](https://github.com/emacs-mirror/emacs/commits?author=Hi-Angel).

**Comment by purcell** [comment]
No worries, there aren't often changes submitted here, so for now I'm going to leave the org members as me, @Fuco1 and @rnikoopour (who had repo-level access previously, and I've now sent an org invite instead). Thanks all!

**Comment by tgross** [comment]
I intended in good faith to be able to help out managing things here but I haven't really had time, or even enough time to ask to be removed. No worries on being removed and sorry I couldn't be of more help!

> I think he's had enough time to show his nature

A wee dramatic, no? :grin:

**Comment by purcell** [comment]
No worries, I always assume good faith by default. I'll try to look at that other PR, @Hi-Angel, just struggling to make time for open source alongside my current consulting work.

**Comment by Hi-Angel** [comment]
@purcell thanks [for review](https://github.com/hcl-emacs/hcl-mode/pull/12) but note that I can't reply there as the discussion is locked 😊

I addressed your comment regarding `font-lock-ensure`.

> and it would be flagged by `package-lint`

It wasn't flagging because `package-lint` isn't linting the tests. I attempted to add them, but ultimately I'm not sure it's a good idea because `package-lint` seems to be specifically purposed for packages and gives warnings like `"Version:" or "Package-Version:" header is missing. MELPA will handle this, but other archives will not`. I presume there's no point adding "Version" header to tests 😊

--------------

I'm a bit confused regarding your notes for makefile:

> I think here I'd probably switch to my fairly-standard Makefile structure, e.g. [as used here](https://github.com/purcell/envrc/blob/master/Makefile)

Could you please highlight the difference, because I was comparing them and the structure looks similar to me

>, and also try to add package-lint to the CI config

It is there 😊 It's in deps to `make all`, so when CI runs `make` everything gets compiled, linted and tested.

> Personal preference would be to not add any code to ensure Emacs 24 compatibility, because that version is super old now.

Thank you, okay, I didn't, instead I dropped 24.3 from the CI in a separate commit. We still test 24 via 24.5.

**Comment by purcell** [comment]
Got it, thanks, that's merged now. 👍 

And yeah, no need to lint the tests.

**Pull request merged**
